### PR TITLE
refactor/829

### LINKED ISSUE #829: Update EDDH (Hamburg Airport)
Requested by marcel510 on Slack

### PR DESCRIPTION
**Pull request by Fechulo**
Resolves #829 

Updates EDDH

### Completion Checklist
- [x] terrain file
- [x] polygonal airspace boundary
- [x] all SIDs
- [x] all STARs
- [x] information to validate traffic levels
- [x] base video map
- [x] no console errors or warnings
- [x] airport guide
---

According to Wikipedia, EDDH has 160,904 yearly operations. Traffic level should fall between 18.4acph and 55.1acph. Currently, there's a total of 19acph 👍

### CODE REVIEW

File: assets/airports/eddh.json [hunk 1]
```diff
@@ -1,844 +1,1435 @@
...
-            "name": "Basum",
-            "suffix": {"05":"1C", "23":"3B", "15":"4D", "33":"1G"},
+        "LUB": {
+            "icao": "LUB",
+            "name": "Lübeck",
+            "suffix": {"05":"9C", "15":"9D", "23":"1B", "33":"9G"},
+            "rwy": {
+                "05": ["HOS"],
+                "15": [["^DH200", "A15+"], "DH201", "HAM"],
+                "23": ["DH111", "DH102", "GT", "DH104"],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: `DH111` is a flyover fix

File: assets/airports/eddh.json [hunk 2]
```diff
@@ -1,844 +1,1435 @@
...
+            ]
+        },
+        "RAMAR": {
+            "icao": "RAMAR",
+            "name": "Ramar",
+            "suffix": {"05":"3C", "15":"3D", "23":"3B", "33":"3G"},
+            "rwy": {
+                "05": ["HOS", "DH154"],
+                "15": [["^DH208", "A23+"], "DH209", "DH212"],
+                "23": ["DH111", "DH102", "GT", "DH104", "DH110"],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: `DH111` is a flyover fix

File: assets/airports/eddh.json [hunk 3]
```diff
@@ -1,844 +1,1435 @@
...
+            "draw": [
+                ["LBV", "WSN*"],
+                ["DH100", "WSN"],
+                ["DH250", "DH254", "LBV"],
+                ["DH216", "WSN"]
+            ]
         }
     },
     "stars": {
         "BOGMU": {
```
[the remark this answers is not in the document]
Comment (Fechulo) [reply assets/airports/eddh.json]: Well it seems like the charts I was using were out of date... :man_facepalming: *sigh*. This is where a Navigraph subscription would come in handy. Looking into it though, these procedures are CDA (Continuous Descent Approach) and "assigned by ATC only", so they would be reserved for emergency aircraft that don't have time to fly normal STAR's. And even in emergency situations, vectors would be used over these procedures, so I don't think adding them is needed. I'll add a note to the airport guide

File: assets/airports/eddh.json [hunk 4]
```diff
@@ -1,844 +1,1435 @@
...
-                "WSN": ["WSN"]
-            },
-            "draw": [["LBV", "WSN"], ["DH100", "WSN"], ["DH250", "DH254", "LBV"], ["DH216", "WSN"]]
+            "draw": [
+                ["DH111", "DH116"],
+                ["DH200", "DH202", "DH216"],
+                ["DH150", "DH153", "LBV", "IDEKO*"]
+            ]
         },
         "LBV": {
```
[the remark this answers is not in the document]
Comment (Fechulo) [reply assets/airports/eddh.json]: elbe
In this case, `Elbe` is the name of the STAR but the ICAO ID is still `LBV`

File: assets/airports/eddh.json [hunk 5]
```diff
@@ -1,844 +1,1435 @@
...
+            "icao": "BASUM",
+            "name": "Basum",
+            "suffix": {"05":"1C", "15":"4D", "23":"3B", "33":"1G"},
             "rwy": {
-                "05": ["HOS", "DH154"],
-                "23": ["DH111", "DH102", "GT", "DH104", "DH110"],
-                "15": [["DH208", "A23+"], "DH209", "DH212"],
-                "33": ["DH250", "DH253", "DH259"]
+                "05": ["DH150", "DH153", "LBV"],
+                "15": [["^DH202", "A20+"], "DH216", "DH221"],
```
[the remark this answers is not in the document]
Comment (Fechulo) [reply assets/airports/eddh.json]: dh202
:thinking: I believe it is

File: assets/airports/eddh.json [hunk 6]
```diff
@@ -1,844 +1,1435 @@
...
+            "name": "Amluh",
+            "suffix": {"05":"8C", "15":"8D", "23":"9B", "33":"8G"},
             "rwy": {
-                "05": ["HOS"],
-                "23": ["DH111", "DH102", "GT", "DH104"],
-                "15": [["DH200", "A15+"], "DH201", "HAM"],
-                "33": ["DH250", "DH253"]
+                "05": ["HOS", "DH151"],
+                "15": [["DH222", "A28+"]],
+                "23": ["DH111", "DH108", "DH115"],
```
[the remark this answers is not in the document]
Comment (Fechulo) [reply assets/airports/eddh.json]: Done 👍

File: assets/airports/eddh.json [hunk 7]
```diff
@@ -1,844 +1,1435 @@
...
         },
         "LBV": {
             "icao": "LBV",
             "name": "Elbe",
-            "suffix": {"05":"1C", "23":"1B", "15":"1D", "33":"1G"},
+            "suffix": {"05":"1C", "15":"1D", "23":"1B", "33":"1G"},
             "rwy": {
                 "05": ["DH150", "DH153"],
+                "15": [["^DH202", "A20+"], "DH216", "DH217"],
                 "23": ["DH111"],
```
[the remark this answers is not in the document]
Comment (Fechulo) [reply assets/airports/eddh.json]: Done 👍

File: assets/airports/eddh.json (hunk 1 above)
Comment (Fechulo) [reply assets/airports/eddh.json]: Done 👍

File: assets/airports/eddh.json (hunk 2 above)
Comment (Fechulo) [reply assets/airports/eddh.json]: Done 👍

File: assets/airports/eddh.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (erikquinn) [reply assets/airports/eddh.json]: image
Hm, mine show that it is not. The date of mine are `26 MAY 17`, so whoever's happen to be more current should do the trick.

File: assets/airports/eddh.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (erikquinn) [reply assets/airports/eddh.json]: Oh god, you're totally right. :man_facepalming: Oops!

File: assets/airports/eddh.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (erikquinn) [reply assets/airports/eddh.json]: That works.

Also, I highly approve of the level of research going on here. Let it be a reminder that _we are better than ATC-SIM!_ :wink:

[the remark this answers is not in the document]
Comment (Fechulo) [reply assets/airports/eddh.json]: 😂 Oh we surpassed those guys a looong time ago!

File: assets/airports/eddh.json [hunk 8]
```diff
@@ -1,844 +1,1435 @@
...
+            "name": "Ekern",
+            "suffix": {"05":"8C", "15":"9D", "23":"8B", "33":"8G"},
             "rwy": {
-                "05": ["HOS", "DH151"],
-                "23": ["DH111", "DH108", "DH115"],
-                "15": [["DH222", "A28+"]],
-                "33": ["DH250", "DH251", "DH252", "HAM"]
+                "05": ["HOS", "DH152"],
+                "15": [["DH200", "A15+"], "DH201", "HAM"],
+                "23": ["DH111", "DH102", "GT", "DH103"],
```
[the remark this answers is not in the document]
Comment (erikquinn) [reply assets/airports/eddh.json]: done

File: assets/airports/eddh.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (erikquinn) [reply assets/airports/eddh.json]: if you're happy with that, then so am I. :smiley:

File: assets/airports/eddh.json [hunk 9]
```diff
@@ -1,844 +1,1435 @@
...
+                "BOGMU": []
             },
-            "body": [["BOGMU", "A100|S250"]],
+            "body": ["BOGMU"],
             "rwy": {
-                "05": [["DH225", "S220"], "DH230", "DH235", "DH255", "DH256", "DH257", "DH025", "ANEXI"],
-                "15": [["DH412", "S220"], "DH416", "DH455", "DH456", "DH457", "DH458"],
-                "23": [["DH634", "S220"], "DH635", "DH655", "DH656", "DH657", "DH658"],
-                "33": [["DH829", "S220"], "DH830", "DH835", "DH854", "DH855", "DH856", "DH857", "DH858"]
+                "05": [["DH225", "S220"], "DH226", "DH227", "DH228", "DH229", ["DH230", "S220"], "DH231", "DH232", "DH233", "DH234", ["DH235", "S220"], "DH255", "DH256", "DH257", "DH025", ["ANEXI", "S170+"]],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: - Where you have `DH025` my charts show `DH258`. They appear to have the same location.
- `ANEXI` appears to be `A30+|S170+`.

File: assets/airports/eddh.json [hunk 10]
```diff
@@ -1,844 +1,1435 @@
...
+                "15": [["DH412", "S220"], "DH413", "DH414", ["DH416", "S220"], "DH455", "DH456", "DH457", "DH458", ["DEVOD", "S170+"]],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: `DEVOD` appears to be `A30+|S170+`

File: assets/airports/eddh.json [hunk 11]
```diff
@@ -1,844 +1,1435 @@
...
+                "23": [["DH634", "S220"], ["DH635", "S220"], "DH655", "DH656", "DH657", "DH658", ["PISAS", "S170+"]],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: `PISAS` appears to be `A30+|S170+`

File: assets/airports/eddh.json [hunk 12]
```diff
@@ -1,844 +1,1435 @@
...
+                "33": [["DH829", "S220"], ["DH830", "S220"], "DH831", "DH832", "DH833", "DH834", ["DH835", "S220"], "DH854", "DH855", "DH856", "DH857", "DH858", ["SOSAX", "S170+"]]
```
Comment (erikquinn) [inline assets/airports/eddh.json]: `SOSAX` appears to be `A30+|S170+`

File: assets/airports/eddh.json [hunk 13]
```diff
@@ -1,844 +1,1435 @@
...
         "BOGMU": {
             "icao": "BOGMU",
             "name": "Bogmu",
+            "suffix": {"05":"05", "15":"15", "23":"23", "33":"33"},
             "entryPoints": {
-                "N850": ["MIC", ["NUSTA", "A170"], ["NIGAD", "A130"]]
+                "BOGMU": []
             },
-            "body": [["BOGMU", "A100|S250"]],
+            "body": ["BOGMU"],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: `BOGMU` appears to be restricted to FL110 or lower.

File: assets/airports/eddh.json [hunk 14]
```diff
@@ -1,844 +1,1435 @@
...
         "RARUP": {
             "icao": "RARUP",
             "name": "Rarup",
+            "suffix": {"05":"05", "15":"15", "23":"23", "33":"33"},
             "entryPoints": {
-                "M748": ["BKD", ["BUMIL", "A230"], ["ABMAL", "A150"]]
+                "RARUP": []
             },
-            "body": [["RARUP", "A100|S250"]],
+            "body": ["RARUP"],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: RARUP appears to be `A110-`

File: assets/airports/eddh.json [hunk 15]
```diff
@@ -1,844 +1,1435 @@
...
             "rwy": {
-                "05": [["DH230", "S220"], "DH235", "DH255", "DH256", "DH257", "DH025", "ANEXI"],
-                "15": [["DH410", "S220"], "DH412", "DH416", "DH455", "DH456", "DH457", "DH458"],
-                "23": [["DH600", "S220"], "DH611", "DH615", "DH655", "DH656", "DH657", "DH658"],
-                "33": [["DH829", "S220"], "DH830", "DH835", "DH854", "DH855", "DH856", "DH857", "DH858"]
+                "05": [["DH230", "S220"], "DH231", "DH232", "DH233", "DH234", ["DH235", "S220"], "DH255", "DH256", "DH257", "DH025", ["ANEXI", "S170+"]],
+                "15": [["DH417", "S220"], ["DH412", "S220"], "DH413", "DH414", ["DH416", "S220"], "DH455", "DH456", "DH457", "DH458", ["DEVOD", "S170+"]],
+                "23": [["DH600", "S220"], ["DH611", "S220"], "DH612", "DH613", "DH614", ["DH615", "S220"], "DH655", "DH656", "DH657", "DH658", ["PISAS", "S170+"]],
+                "33": [["DH829", "S220"], ["DH830", "S220"], "DH831", "DH832", "DH833", "DH834", ["DH835", "S220"], "DH854", "DH855", "DH856", "DH857", "DH858", ["SOSAX", "S170+"]]
```
Comment (erikquinn) [inline assets/airports/eddh.json]: Same comment on RARUP STAR FAFs being `A30+|S170+`

File: assets/airports/eddh.json [hunk 16]
```diff
@@ -1,844 +1,1435 @@
...
+            "body": ["NOLGO"],
             "rwy": {
-                "05": [["DH230", "S220"], "DH235", "DH255", "DH256", "DH257", "DH025", "ANEXI"],
-                "15": ["DH406", ["DH412", "S220"], "DH416", "DH455", "DH456", "DH457", "DH458"],
-                "23": [["DH611", "S220"], "DH615", "DH655", "DH656", "DH657", "DH658"],
-                "33": ["DH832", ["DH835", "S220"], "DH854", "DH855", "DH856", "DH857", "DH858"]
+                "05": [["DH230", "S220"], "DH231", "DH232", "DH233", "DH234", ["DH235", "S220"], "DH255", "DH256", "DH257", "DH025", ["ANEXI", "S170+"]],
+                "15": ["DH406", "DH407", "DH408", "DH409", ["DH412", "S220"], "DH413", "DH414", ["DH416", "S220"], "DH455", "DH456", "DH457", "DH458", ["DEVOD", "S170+"]],
+                "23": [["DH611", "S220"], "DH612", "DH613", "DH614", ["DH615", "S220"], "DH655", "DH656", "DH657", "DH658", ["PISAS", "S170+"]],
+                "33": ["DH832", "DH833", "DH834", ["DH835", "S220"], "DH854", "DH855", "DH856", "DH857", "DH858", ["SOSAX", "S170+"]]
```
Comment (erikquinn) [inline assets/airports/eddh.json]: Same comment on NOLGO STAR FAFs being `A30+|S170+`

File: assets/airports/eddh.json [hunk 17]
```diff
@@ -1,844 +1,1435 @@
...
+                "RARUP": []
             },
-            "body": [["RARUP", "A100|S250"]],
+            "body": ["RARUP"],
             "rwy": {
-                "05": [["DH230", "S220"], "DH235", "DH255", "DH256", "DH257", "DH025", "ANEXI"],
-                "15": [["DH410", "S220"], "DH412", "DH416", "DH455", "DH456", "DH457", "DH458"],
-                "23": [["DH600", "S220"], "DH611", "DH615", "DH655", "DH656", "DH657", "DH658"],
-                "33": [["DH829", "S220"], "DH830", "DH835", "DH854", "DH855", "DH856", "DH857", "DH858"]
+                "05": [["DH230", "S220"], "DH231", "DH232", "DH233", "DH234", ["DH235", "S220"], "DH255", "DH256", "DH257", "DH025", ["ANEXI", "S170+"]],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: Where you have `DH025`, I show `DH258`, though they appear to have the same coordinates.

File: assets/airports/eddh.json [hunk 18]
```diff
@@ -1,844 +1,1435 @@
...
         "NOLGO": {
             "icao": "NOLGO",
             "name": "Nolgo",
+            "suffix": {"05":"05", "15":"15", "23":"23", "33":"33"},
             "entryPoints": {
-                "P605": ["DLE", ["MADUB", "A280"], ["AGATI", "A200"]]
+                "NOLGO": []
             },
-            "body": [["NOLGO", "A100|S250"]],
+            "body": ["NOLGO"],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: NOLGO appears to be `A110-`

File: assets/airports/eddh.json (hunk 16 above)
Comment (erikquinn) [inline assets/airports/eddh.json]: on 33 transition, `DH832` appears to be `S220`

File: assets/airports/eddh.json [hunk 19]
```diff
@@ -1,844 +1,1435 @@
...
             "rwy": {
-                "05": [["DH230", "S220"], "DH235", "DH255", "DH256", "DH257", "DH025", "ANEXI"],
-                "15": ["DH406", ["DH412", "S220"], "DH416", "DH455", "DH456", "DH457", "DH458"],
-                "23": [["DH611", "S220"], "DH615", "DH655", "DH656", "DH657", "DH658"],
-                "33": ["DH832", ["DH835", "S220"], "DH854", "DH855", "DH856", "DH857", "DH858"]
+                "05": [["DH230", "S220"], "DH231", "DH232", "DH233", "DH234", ["DH235", "S220"], "DH255", "DH256", "DH257", "DH025", ["ANEXI", "S170+"]],
+                "15": ["DH406", "DH407", "DH408", "DH409", ["DH412", "S220"], "DH413", "DH414", ["DH416", "S220"], "DH455", "DH456", "DH457", "DH458", ["DEVOD", "S170+"]],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: `DH406` appears to be `S220`

File: assets/airports/eddh.json [hunk 20]
```diff
@@ -1,844 +1,1435 @@
...
+            "body": ["RIBSO"],
             "rwy": {
-                "05": [["DH210", "S220"], "DH211", "DH213", "DH215", "DH255", "DH256", "DH257", "DH025", "ANEXI"],
-                "15": [["DH434", "S220"], "DH435", "DH455", "DH456", "DH457", "DH458"],
-                "23": ["DH620", ["DH633", "S220"], "DH634", "DH635", "DH655", "DH656", "DH657", "DH658"],
-                "33": [["DH804", "S220"], "DH815", "DH854", "DH855", "DH856", "DH857", "DH858"]
+                "05": [["DH210", "S220"], "DH211", "DH213", "DH214", ["DH215", "S220"], "DH255", "DH256", "DH257", "DH025", ["ANEXI", "S170+"]],
+                "15": [["DH434", "S220"], ["DH435", "S220"], "DH455", "DH456", "DH457", "DH458", ["DEVOD", "S170+"]],
+                "23": ["DH620", ["DH633", "S220"], ["DH634", "S220"], ["DH635", "S220"], "DH655", "DH656", "DH657", "DH658", ["PISAS", "S170+"]],
+                "33": [["DH804", "S220"], "DH805", "DH806", "DH807", "DH808", "DH809", "DH810", "DH811", "DH812", "DH813", "DH814", ["DH815", "S220"], "DH854", "DH855", "DH856", "DH857", "DH858", ["SOSAX", "S170+"]]
```
Comment (erikquinn) [inline assets/airports/eddh.json]: Same comment on RIBSO STAR FAFs being `A30+|S170+`

File: assets/airports/eddh.json [hunk 21]
```diff
@@ -1,844 +1,1435 @@
...
         "RIBSO": {
             "icao": "RIBSO",
             "name": "Ribso",
+            "suffix": {"05":"05", "15":"15", "23":"23", "33":"33"},
             "entryPoints": {
-                "L619": ["DHE", ["GOLEN", "A220"], ["OSTOR", "A120"]]
+                "RIBSO": []
             },
-            "body": [["RIBSO", "A100|S250"]],
+            "body": ["RIBSO"],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: RIBSO appears to be `A110-`.

File: assets/airports/eddh.json [hunk 22]
```diff
@@ -1,844 +1,1435 @@
...
             "rwy": {
-                "05": [["DH210", "S220"], "DH211", "DH213", "DH215", "DH255", "DH256", "DH257", "DH025", "ANEXI"],
-                "15": [["DH434", "S220"], "DH435", "DH455", "DH456", "DH457", "DH458"],
-                "23": ["DH620", ["DH633", "S220"], "DH634", "DH635", "DH655", "DH656", "DH657", "DH658"],
-                "33": [["DH804", "S220"], "DH815", "DH854", "DH855", "DH856", "DH857", "DH858"]
+                "05": [["DH210", "S220"], "DH211", "DH213", "DH214", ["DH215", "S220"], "DH255", "DH256", "DH257", "DH025", ["ANEXI", "S170+"]],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: - `DH213` apepars to be `S220`
- Where you have `DH025`, I show `DH258`, though they appear to have the same coordinates.

File: assets/airports/eddh.json [hunk 23]
```diff
@@ -1,844 +1,1435 @@
...
+                "15": [["DH434", "S220"], ["DH435", "S220"], "DH455", "DH456", "DH457", "DH458", ["DEVOD", "S170+"]],
+                "23": ["DH620", ["DH633", "S220"], ["DH634", "S220"], ["DH635", "S220"], "DH655", "DH656", "DH657", "DH658", ["PISAS", "S170+"]],
```
Comment (erikquinn) [inline assets/airports/eddh.json]: I show no speed at `DH634`

**Comment by Fechulo** [comment]
@erikquinn I should've probably said this earlier, oops 😬 The altitude restrictions for the STAR's in Germany are similar to the "Expect x altitude" in the US, so ATC is still expected to give descent clearances. Speed restrictions still apply though, so I'll take a look at those

**Comment by erikquinn** [comment]
Gotcha, looks like you've done the rest, so we should be good to go!

**Pull request merged**